lib/sinatra/base.rb

```diff
@@ -1069,7 +1081,7 @@ def static!(options = {})
     def indifferent_params(object)
       case object
       when Hash
-        new_hash = indifferent_hash
+        new_hash = IndifferentHash.new
```

Comment:
I'd suggest also use `each_with_object` to loop the keys+values, instead of create an variable, and then return that variable:
```ruby
object.each_with_object(IndifferentHash.new) do  |(key, value), new_hash|
  new_hash[key] = indifferent_params(value)
end
```